src/apps/testapps/testPolygon.c

```diff
@@ -237,4 +237,85 @@ TEST(bboxFromLinkedGeoLoopNoVertices) {
     destroyLinkedGeoLoop(&loop);
 }
 
+TEST(isClockwiseGeofence) {
+    GeoCoord verts[] = {{0, 0}, {0.1, 0.1}, {0, 0.1}};
+
+    Geofence geofence;
+    geofence.verts = verts;
+    geofence.numVerts = 3;
+
+    t_assert(isClockwiseGeofence(&geofence) == true,
+             "Got true for clockwise geofence");
+}
+
+TEST(isClockwiseLinkedGeoLoop) {
+    const GeoCoord verts[] = {{0.1, 0.1}, {0.2, 0.2}, {0.1, 0.2}};
+
+    LinkedGeoLoop loop;
+    initLinkedLoop(&loop);
+
+    for (int i = 0; i < 3; i++) {
+        addLinkedCoord(&loop, &verts[i]);
+    }
+
+    t_assert(isClockwiseLinkedGeoLoop(&loop) == true,
+             "Got true for clockwise loop");
```

Comment:
There are various points of view on assertion messages:
- Print what happened in the success case: `Got true for clockwise loop`
- Print what happened in the failure case: `Got false for clockwise loop`
- Print the requirement: `Clockwise loop should return true`
- Print a description of what's under test: `Clockwise loop`

These messages are valuable when tests fail, but I tend to think that the more common role of tests is as spec documentation, and these messages are going to be most frequently read by other developers looking into the function under test or adding additional tests. The "failure case" option is the worst for this use case, because it states what should not happen, not what should happen, making tests-as-spec practically useless.

I usually think the "requirement" option is the best for both use cases, especially if you have a nicer test framework that pretty prints successes and failures, so you see a list of passing and failing requirements. But the general style in this repo is the "success case" version, so I've used that here.

Reply:
OK I see. Sort of like BDD style ("it should do x").